vault/token_store.go

```diff
@@ -1325,32 +1339,58 @@ func (ts *TokenStore) handleUpdateLookupAccessor(req *logical.Request, data *fra
 // the token associated with the accessor
 func (ts *TokenStore) handleUpdateRevokeAccessor(req *logical.Request, data *framework.FieldData) (*logical.Response, error) {
 	var urlaccessor bool
-	accessor := data.Get("accessor").(string)
-	if accessor == "" {
-		accessor = data.Get("urlaccessor").(string)
-		if accessor == "" {
+	accessors := data.Get("accessor").([]string)
+
+	if len(accessors) == 0 {
+		accessors = data.Get("urlaccessor").([]string)
+		if len(accessors) == 0 {
 			return nil, &logical.StatusBadRequest{Err: "missing accessor"}
 		}
 		urlaccessor = true
 	}
 
-	aEntry, err := ts.lookupByAccessor(accessor, true)
-	if err != nil {
-		return nil, err
+	errs := make([]error, len(accessors))
+	tokens := make([]string, len(accessors))
+
+	for idx, accessor := range accessors {
+		aEntry, err := ts.lookupByAccessor(accessor, true)
+		if err != nil {
+			errs[idx] = err
+			tokens[idx] = ""
+		}
+
+		tokens[idx] = aEntry.TokenID
 	}
 
-	// Revoke the token and its children
-	if err := ts.RevokeTree(aEntry.TokenID); err != nil {
-		return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
+	revokeErrors := ts.RevokeTrees(tokens)
+
+	response := &logical.Response{}
+	failedRevokes := make([]map[string]string, 0, len(revokeErrors))
+
+	for idx, revokeError := range revokeErrors {
+		if errs[idx] == nil {
+			errs[idx] = revokeError
+		}
+
+		if errs[idx] != nil {
+			failedRevokes = append(failedRevokes, map[string]string{
+				"accessor": accessors[idx],
+				"error":    errs[idx].Error(),
+			})
+		}
+	}
+
+	if len(failedRevokes) > 0 {
+		response.SetError(fmt.Errorf("contains failed revokes"), failedRevokes)
 	}
 
 	if urlaccessor {
-		resp := &logical.Response{}
-		resp.AddWarning(`Using an accessor in the path is unsafe as the accessor can be logged in many places. Please use POST or PUT with the accessor passed in via the "accessor" parameter.`)
-		return resp, nil
+		response.AddWarning(`Using an accessor in the path is unsafe as the accessor can be logged in many places. Please use POST or PUT with the accessor passed in via the "accessor" parameter.`)
+	} else if len(failedRevokes) == 0 {
+		return nil, nil
 	}
 
-	return nil, nil
```

Comment:
If only a single accessor is given, the previous return values should be retained.